### DISCUSSION TITLE
TOTP Auto-Submit conflicts with Last-Pass Auto Fill and Locks Me Out

### DISCUSSION THREAD

**Discussion by djha-skin** [community/community#70489] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

# The Bad User Experience

Here is what happens:

1. I log into github with my username and password.
2. I am presented with a TOTP field.
3. My LastPass browser plugin says "Oh! I know! That's a PASSWORD field! I'll auto-fill your github password there!" and it does so.
4. The GitHub page, also trying to be helpful, says "Oh! You've entered your code already! I'll just submit your request then!"
5. The code is incorrect, as codes have nothing to do with the password (and LastPass fills in the wrong account stuff anyways).
6. GitHub again presents me with the TOTP page, this time with a "Incorrect Code" banner.
7. LastPass, ever helpful, auto-fills again.
8. GitHub, also eager, auto-submits.

This process repeats about 10 times within the span of _half a second_, after which I am informed that, since it looks like a bot is trying to break into my account -- which, in fairness, it is, kind of -- I'll need to wait "a while" before I can log in. This usually means "Come back tomorrow" given the number of times the log-in attempts occurred.

I use GitHub professionally -- my company has a paid account. This means I can't get any work done for the rest of the day.

As a workaround, I have had to uninstall the browser plugin for LastPass, but I have to install it sometimes anyways or I can't get into some of the sites I need for work.

# The Problem

The main problem is that these two applications -- LastPass and GitHub -- are both trying to be helpful, but in ways I cannot turn off if they are harmful.

It is reasonable to assume that a password manager should autofill; that's its job.

It is _not_ reasonable for a website to auto-submit a form the fields of which I am still in the middle of editing. I need time to be able to correct LastPass's mistake before hitting "Submit".

# The Ask

GitHub needs to have an option to turn off this auto-submit behavior on TOTP logins so that I can still use my work password manager and use TOTP at the same time.

**Comment by DNin01** [comment]
Yeah, that's frustrating.

There might be an option to turn off autofill in LastPass (I don't know - I use a different password manager). Or maybe you could try having your extensions page ready in another tab so you can quickly turn off the LastPass browser extension after your password is autofilled?

To keep this from happening again, try [setting a different authentication option](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/changing-your-preferred-two-factor-authentication-method) (such as GitHub Mobile or a security key) [as preferred](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/changing-your-preferred-two-factor-authentication-method), once you're logged back in. Using 2FA with GitHub Mobile or a security key shouldn't bring up a code field on your browser.

**Reply by djha-skin** [reply]
Thank you for your reply.

I was able to get LastPass to stop autofilling, it does have an option.

> as preferred

I very much like TOTP, as it is very password-manager-friendly. My password manager stores all my TOTP keys. I use it for everything. It is my fervent hope that support for it continues far into the future :)

I tried a yubi key, but they are so much more unusable compared to just hitting the "fill TOTP" button. (I use KeePassXC for my personal passwords.) And then I have my TOTP stuff everywhere.

It's interesting to me that the security key  is more preferred, thanks for the link.

At any rate: this does, I think, deserve attention. I hope that this is changed to an option or fixed.

**Reply by djha-skin** [reply]
Ah, I see -- as *I* prefer. Just read the link, sorry for the confusion.
